### PR TITLE
Document script discovery wrt changedir

### LINKED ISSUE #2089: changedir breaks script discovery
Hi, firstly thanks for providing Tox. It's awesome! I also saw the PyCon talk about Tox 4 and I'm really looking forward to it 😄

I've created an env for building Sphinx documentation with its make scripts. I know the documentation suggests using `sphinx-build` directly but anyways: the script cannot be found under the directory.

> InvocationError for command could not find executable my_script

<details>
<summary>

Output of `tox -e script -rvv`

</summary>

```sh
(venv) C:\Felix\Git\ztst>tox -e script -rvv
using tox.ini: C:\Felix\Git\ztst\tox.ini (pid 14348)
  removing C:\Felix\Git\ztst\.tox\log
using tox-3.23.1 from c:\felix\git\ztst\venv\lib\site-packages\tox\__init__.py (pid 14348)
GLOB start: packaging 
GLOB sdist-make: C:\Felix\Git\ztst\setup.py
  removing C:\Felix\Git\ztst\.tox\dist
[6856] C:\Felix\Git\ztst$ 'c:\felix\git\ztst\venv\scripts\python.exe' setup.py sdist --formats=zip --dist-dir '.tox\dist' >.tox\log
\GLOB-0.log
running sdist
running egg_info
writing package.egg-info\PKG-INFO
writing dependency_links to package.egg-info\dependency_links.txt
writing top-level names to package.egg-info\top_level.txt
reading manifest file 'package.egg-info\SOURCES.txt'
writing manifest file 'package.egg-info\SOURCES.txt'
warning: sdist: standard file not found: should have one of README, README.rst, README.txt, README.md

running check
warning: check: missing required meta-data: url

creating package-0.0.1
creating package-0.0.1\package
creating package-0.0.1\package.egg-info
copying files to package-0.0.1...
copying setup.py -> package-0.0.1
copying package\__init__.py -> package-0.0.1\package
copying package.egg-info\PKG-INFO -> package-0.0.1\package.egg-info
copying package.egg-info\SOURCES.txt -> package-0.0.1\package.egg-info
copying package.egg-info\dependency_links.txt -> package-0.0.1\package.egg-info
copying package.egg-info\top_level.txt -> package-0.0.1\package.egg-info
Writing package-0.0.1\setup.cfg
creating '.tox\dist\package-0.0.1.zip' and adding 'package-0.0.1' to it
adding 'package-0.0.1'
adding 'package-0.0.1\package'
adding 'package-0.0.1\package.egg-info'
adding 'package-0.0.1\PKG-INFO'
adding 'package-0.0.1\setup.cfg'
adding 'package-0.0.1\setup.py'
adding 'package-0.0.1\package\__init__.py'
adding 'package-0.0.1\package.egg-info\dependency_links.txt'
adding 'package-0.0.1\package.egg-info\PKG-INFO'
adding 'package-0.0.1\package.egg-info\SOURCES.txt'
adding 'package-0.0.1\package.egg-info\top_level.txt'
removing 'package-0.0.1' (and everything under it)

GLOB finish: packaging  after 0.28 seconds
copying new sdistfile to 'C:\\Users\\Felix The III\\.tox\\distshare\\package-0.0.1.zip'
package .tmp\package\1\package-0.0.1.zip links to dist\package-0.0.1.zip (C:\Felix\Git\ztst\.tox)
c:\felix\git\ztst\venv\scripts\python.exe (c:\felix\git\ztst\venv\scripts\python.exe) is {'executable': 'c:\\felix\\git\\ztst\\venv
\\scripts\\python.exe', 'implementation': 'CPython', 'version_info': [3, 9, 4, 'final', 0], 'version': '3.9.4 (tags/v3.9.4:1f2e308,
 Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)]', 'is_64': True, 'sysplatform': 'win32', 'os_sep': '\\', 'extra_version_info':
None}
script uses c:\felix\git\ztst\venv\scripts\python.exe
script start: getenv C:\Felix\Git\ztst\.tox\script
script cannot reuse: -r flag
script recreate: C:\Felix\Git\ztst\.tox\script
  removing C:\Felix\Git\ztst\.tox\script
setting PATH=C:\Felix\Git\ztst\.tox\script\Scripts;C:\Felix\Git\ztst\venv\Scripts;C:\Program Files\Python39\Scripts\;C:\Program Fil
es\Python39\;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin;C:\Program Files\R\Rtools\bin;C:\Program Files\Python37\S
cripts\;C:\Program Files\Python37\;C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;C:\Program Fi
les (x86)\Common Files\Intel\Shared Libraries\redist\ia32\compiler;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windo
ws\System32\WindowsPowerShell\v1.0\;C:\Users\Felix The III\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft
 SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Program Files (x86)\NVIDIA Corp
oration\PhysX\Common;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files (x86)\WinSCP\;C:\WINDOWS\system32;C:\WINDOWS;C:\W
INDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Meld;C:\Progra
m Files\Java\jdk1.8.0_131\jre\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2019.5.0\;C:\Ruby26-x64\Perl\c\bin;C:\Ruby26-x
64\Perl\perl\site\bin;C:\Ruby26-x64\Perl\perl\bin;C:\Program Files\MATLAB\R2020a\bin;C:\Program Files\PuTTY\;C:\Program Files\Inksc
ape\bin;C:\Program Files\dotnet\;C:\Program Files (x86)\Graphviz\bin;C:\Program Files\Git\bin;;C:\WINDOWS\system32;C:\WINDOWS;C:\WI
NDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Ruby26-x64\bin;C:\Users\Felix The I
II\AppData\Local\Microsoft\WindowsApps;C:\Users\Felix The III\AppData\Local\Microsoft\WindowsApps
[12584] C:\Felix\Git\ztst\.tox$ 'c:\felix\git\ztst\venv\scripts\python.exe' -m virtualenv --no-download --python 'c:\felix\git\ztst
\venv\scripts\python.exe' script
created virtual environment CPython3.9.4.final.0-64 in 443ms
  creator CPython3Windows(dest=C:\Felix\Git\ztst\.tox\script, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\Felix The III\App
Data\Local\pypa\virtualenv)
    added seed packages: pip==21.1.2, setuptools==57.0.0, wheel==0.36.2
  activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
script finish: getenv C:\Felix\Git\ztst\.tox\script after 0.89 seconds
script start: installpkg C:\Felix\Git\ztst\.tox\.tmp\package\1\package-0.0.1.zip
script inst: C:\Felix\Git\ztst\.tox\.tmp\package\1\package-0.0.1.zip
write config to C:\Felix\Git\ztst\.tox\script\.tox-config1 as 'a1b7ef8ecc300c0c3968e75eee0bacebceafee09837fc27d41da46c09ea194b3 c:\
\felix\\git\\ztst\\venv\\scripts\\python.exe\n3.23.1 0 0 0'
setting PATH=C:\Felix\Git\ztst\.tox\script\Scripts;C:\Felix\Git\ztst\venv\Scripts;C:\Program Files\Python39\Scripts\;C:\Program Fil
es\Python39\;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin;C:\Program Files\R\Rtools\bin;C:\Program Files\Python37\S
cripts\;C:\Program Files\Python37\;C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;C:\Program Fi
les (x86)\Common Files\Intel\Shared Libraries\redist\ia32\compiler;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windo
ws\System32\WindowsPowerShell\v1.0\;C:\Users\Felix The III\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft
 SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Program Files (x86)\NVIDIA Corp
oration\PhysX\Common;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files (x86)\WinSCP\;C:\WINDOWS\system32;C:\WINDOWS;C:\W
INDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Meld;C:\Progra
m Files\Java\jdk1.8.0_131\jre\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2019.5.0\;C:\Ruby26-x64\Perl\c\bin;C:\Ruby26-x
64\Perl\perl\site\bin;C:\Ruby26-x64\Perl\perl\bin;C:\Program Files\MATLAB\R2020a\bin;C:\Program Files\PuTTY\;C:\Program Files\Inksc
ape\bin;C:\Program Files\dotnet\;C:\Program Files (x86)\Graphviz\bin;C:\Program Files\Git\bin;;C:\WINDOWS\system32;C:\WINDOWS;C:\WI
NDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Ruby26-x64\bin;C:\Users\Felix The I
II\AppData\Local\Microsoft\WindowsApps;C:\Users\Felix The III\AppData\Local\Microsoft\WindowsApps
[6656] C:\Felix\Git\ztst$ 'C:\Felix\Git\ztst\.tox\script\Scripts\python.EXE' -m pip install --exists-action w '.tox\.tmp\package\1\
package-0.0.1.zip'
Processing c:\felix\git\ztst\.tox\.tmp\package\1\package-0.0.1.zip
Building wheels for collected packages: package
  Building wheel for package (setup.py) ... done
  Created wheel for package: filename=package-0.0.1-py3-none-any.whl size=1299 sha256=2b23f54312efcb9ccde8fdc40a2e0bde06b6699d9c261
ddec7add086f590c4b9
  Stored in directory: c:\users\felix the iii\appdata\local\pip\cache\wheels\3a\ff\d7\7610bc53c7eb3c2f1bbde735f922ecaad85b79d792533
8629c
Successfully built package
Installing collected packages: package
Successfully installed package-0.0.1
script finish: installpkg C:\Felix\Git\ztst\.tox\.tmp\package\1\package-0.0.1.zip after 2.81 seconds
script start: envreport 
setting PATH=C:\Felix\Git\ztst\.tox\script\Scripts;C:\Felix\Git\ztst\venv\Scripts;C:\Program Files\Python39\Scripts\;C:\Program Fil
es\Python39\;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin;C:\Program Files\R\Rtools\bin;C:\Program Files\Python37\S
cripts\;C:\Program Files\Python37\;C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;C:\Program Fi
les (x86)\Common Files\Intel\Shared Libraries\redist\ia32\compiler;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windo
ws\System32\WindowsPowerShell\v1.0\;C:\Users\Felix The III\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft
 SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Program Files (x86)\NVIDIA Corp
oration\PhysX\Common;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files (x86)\WinSCP\;C:\WINDOWS\system32;C:\WINDOWS;C:\W
INDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Meld;C:\Progra
m Files\Java\jdk1.8.0_131\jre\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2019.5.0\;C:\Ruby26-x64\Perl\c\bin;C:\Ruby26-x
64\Perl\perl\site\bin;C:\Ruby26-x64\Perl\perl\bin;C:\Program Files\MATLAB\R2020a\bin;C:\Program Files\PuTTY\;C:\Program Files\Inksc
ape\bin;C:\Program Files\dotnet\;C:\Program Files (x86)\Graphviz\bin;C:\Program Files\Git\bin;;C:\WINDOWS\system32;C:\WINDOWS;C:\WI
NDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Ruby26-x64\bin;C:\Users\Felix The I
II\AppData\Local\Microsoft\WindowsApps;C:\Users\Felix The III\AppData\Local\Microsoft\WindowsApps
[12348] C:\Felix\Git\ztst$ 'C:\Felix\Git\ztst\.tox\script\Scripts\python.EXE' -m pip freeze >.tox\script\log\script-0.log
script finish: envreport  after 0.40 seconds
script installed: package @ file:///C:/Felix/Git/ztst/.tox/.tmp/package/1/package-0.0.1.zip
script start: run-test-pre 
script run-test-pre: PYTHONHASHSEED='551'
script finish: run-test-pre  after 0.00 seconds
script start: run-test 
script run-test: commands[0] | my_script
setting PATH=C:\Felix\Git\ztst\.tox\script\Scripts;C:\Felix\Git\ztst\venv\Scripts;C:\Program Files\Python39\Scripts\;C:\Program Fil
es\Python39\;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin;C:\Program Files\R\Rtools\bin;C:\Program Files\Python37\S
cripts\;C:\Program Files\Python37\;C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;C:\Program Fi
les (x86)\Common Files\Intel\Shared Libraries\redist\ia32\compiler;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windo
ws\System32\WindowsPowerShell\v1.0\;C:\Users\Felix The III\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft
 SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Program Files (x86)\NVIDIA Corp
oration\PhysX\Common;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files (x86)\WinSCP\;C:\WINDOWS\system32;C:\WINDOWS;C:\W
INDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Meld;C:\Progra
m Files\Java\jdk1.8.0_131\jre\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2019.5.0\;C:\Ruby26-x64\Perl\c\bin;C:\Ruby26-x
64\Perl\perl\site\bin;C:\Ruby26-x64\Perl\perl\bin;C:\Program Files\MATLAB\R2020a\bin;C:\Program Files\PuTTY\;C:\Program Files\Inksc
ape\bin;C:\Program Files\dotnet\;C:\Program Files (x86)\Graphviz\bin;C:\Program Files\Git\bin;;C:\WINDOWS\system32;C:\WINDOWS;C:\WI
NDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Ruby26-x64\bin;C:\Users\Felix The I
II\AppData\Local\Microsoft\WindowsApps;C:\Users\Felix The III\AppData\Local\Microsoft\WindowsApps
ERROR: InvocationError for command could not find executable my_script
script finish: run-test  after 0.04 seconds
script start: run-test-post 
script finish: run-test-post  after 0.00 seconds
____________________________________________________________ summary _____________________________________________________________
ERROR:   script: commands failed
cleanup C:\Felix\Git\ztst\.tox\.tmp\package\1\package-0.0.1.zip
```

</details>

Environment: Windows 10, Python 3.9.4, virtualenv, tox 3.23.1

```sh
; my_script.bat
echo YEAH.
```

```ini
; tox.ini
[testenv:script]
changedir = doc
allowlist_externals = my_script
commands = my_script
```

I also had a dummy project with a `setup.py` file just for installing the package.

- When the script is in the top level folder (and there's no changedir), this works
- This works on Linux ([conf](https://github.com/felix-hilden/tekore/blob/master/tox.ini#L43), [workflow](https://github.com/felix-hilden/tekore/blob/master/.github/workflows/build.yml#L41), [successful run](https://github.com/felix-hilden/tekore/runs/2822036690?check_suite_focus=true))
- I'm *pretty sure* this has worked on Windows before

For the reasons above, I posted this as a bug. But feel free to correct me if I'm simply doing something wrong!

### PR DESCRIPTION
**Pull request by felix-hilden**
Closes #2089: This PR documents the special behavior of `changedir` with batch / shell script discovery.

- [x] wrote descriptive pull request text
- [x] (N/A) added/updated test(s)
- [x] updated/extended the documentation
- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [x] (N/A) added news fragment in [changelog folder](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

Possible review points:
- The two back-to-back notes in the config documentation look a bit ugly, but I think it's not appropriate to combine them either, because they talk about two different things
- Something to improve / change about the documentation content itself?

### CODE REVIEW

File: docs/example/basic.rst [hunk 1]
```diff
@@ -114,6 +114,33 @@ configuration:
 
 .. _virtualenv: https://pypi.org/project/virtualenv
 
+.. _script-command:
+
+Running scripts as commands
+----------------------------------------------
+
+Shell and batch scripts can be run like other commands with one caveat:
```
Comment (gaborbernat) [inline docs/example/basic.rst:122]: This should be added under the commands configuration, not here 👍

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: docs/example/basic.rst (hunk 1 above)
Comment (felix-hilden) [reply docs/example/basic.rst:122]: Just to make sure I understood correctly, the whole text should be in the `changedir` conf section?

Comment (gaborbernat) [reply docs/example/basic.rst:122]: No, it should be under https://tox.readthedocs.io/en/latest/config.html#conf-commands

Comment (felix-hilden) [reply docs/example/basic.rst:122]: Alright, I see it now! I'll be available to work on this later today.

Comment (felix-hilden) [reply docs/example/basic.rst]: It is now in the commands section!

**Comment by felix-hilden** [comment]
Just a friendly bump incase it went unnoticed because of the force push 😅 ready for another round of review now!

**Comment by jugmac00** [comment]
Hi @felix-hilden - I do not want to spoil @gaborbernat's fun to review this PR, so, just a note from my side: Most of the CI is failing, and it looks like this PR caused the failing test(s).
see https://dev.azure.com/toxdev/tox/_test/analytics?definitionId=9&contextType=build

Could you have a look and try to get CI green?

I am not sure whether you have access to the build logs of azure - just in case, you can run the tests locally with... `tox` :-)

**Comment by felix-hilden** [comment]
Oh, thanks @jugmac00 😅 I can see the results, but I was a bit overwhelmed with the dev setup since I only have Python 3.9 installed, so I fiddled with the tox ini manually and just ran doc builds. And then when submitting the PR I just decided to ignore the failures. Sorry about that. I'll try to fix it right away!

File: docs/config.rst [hunk 2]
```diff
@@ -273,6 +273,23 @@ Complete list of settings that you can put into ``testenv*`` sections:
         translates as the virtual environments ``python`` (having the same runtime version
         as the :conf:`basepython`), and ``pip`` translates as the virtual environments ``pip``.
 
+    :note: Shell and batch scripts can be run like other commands with one caveat:
```
Comment (gaborbernat) [inline docs/config.rst]: This is a bit too verbose and specific for my test so made it a bit simpler.

**Review by gaborbernat — APPROVED** [review]
Thanks for your contribution!

**Comment by felix-hilden** [comment]
Uhm, you're welcome and all, but did you force push the branch to be empty and close the pull request without merging? Am I missing something about the contributing flow?

**Comment by gaborbernat** [comment]
Github messed up here, when I pushed my changes it close the PR... without pushing the commits. Opened now under https://github.com/tox-dev/tox/pull/2096/files

**Pull request closed without merging**